### PR TITLE
Remove useless assignments

### PR DESCRIPTION
**Pull request by IgorPolyakov**
Hey! this is my first `crystal` contribution. I ran the source code through the ameba analyzer (ttps://github.com/crystal-ameba/ameba) and fixed a few lines.

If it's helpful I can correct others comments.

The pull request is in the draft stage, if anything needs to be fixed, I'd love some feedback.

### CODE REVIEW

File: src/mime/media_type.cr [hunk 1]
```diff
@@ -447,8 +447,8 @@ module MIME
     end
 
     private def self.decode_rfc2231(encoded : String)
-      encoding, split1, rest = encoded.partition('\'')
-      lang, split2, value = rest.partition('\'')
+      encoding, _split_one, rest = encoded.partition('\'')
+      _lang, _split_two, value = rest.partition('\'')
```
Comment (straight-shoota) [inline src/mime/media_type.cr:451]: I think we can just use underscore without names to make the relevant variable names pop out more.
```suggestion
      encoding, _, rest = encoded.partition('\'')
      _, _, value = rest.partition('\'')
```
Maybe keep `_lang` as a label, but I'd certainly toss out the separator names.

**Review by straight-shoota — APPROVED** [review]
LGTM, just a minor suggestion.

File: src/mime/media_type.cr (hunk 1 above)
Comment (HertzDevil) [reply src/mime/media_type.cr:451]: `_lang` would still be unused. Does Ameba intentionally skip over these variables? (I know that Elixir does, for example.)

Comment (straight-shoota) [reply src/mime/media_type.cr:451]: Yes, I think the purpose of variable names with underscore prefix is to indicate that variable is unused, but still give it a label to document context.

Comment (IgorPolyakov) [reply src/mime/media_type.cr]: I fixed `split` word, and use the lang with the `_` prefix, linter doesn't worry about this line now

**Review by straight-shoota — APPROVED** [review]

**Review by oprypin — APPROVED** [review]

**Pull request merged**